backend/src/database/repositories/memberAttributeSettingsRepository.ts

```diff
@@ -0,0 +1,230 @@
+import Error404 from '../../errors/Error404'
+import SequelizeFilterUtils from '../utils/sequelizeFilterUtils'
+
+const Op = Sequelize.Op
+
+class MemberAttributeSettingsRepository {
+  static async findById(id, options: IRepositoryOptions) {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const include = []
+
+    const currentTenant = SequelizeRepository.getCurrentTenant(options)
+
+    const record = await options.database.memberAttributeSettings.findOne({
+      where: {
+        id,
+        tenantId: currentTenant.id,
+      },
+      include,
+      transaction,
+    })
+
+    if (!record) {
+      throw new Error404()
+    }
+
+    return this._populateRelations(record)
+  }
+
+  static async _populateRelations(record) {
+    if (!record) {
+      return record
+    }
+    const output = record.get({ plain: true })
+
+    return output
+  }
+
+  static async create(data, options: IRepositoryOptions) {
+    const currentUser = SequelizeRepository.getCurrentUser(options)
+
+    const tenant = SequelizeRepository.getCurrentTenant(options)
+
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const record = await options.database.memberAttributeSettings.create(
+      {
+        ...lodash.pick(data, ['type', 'name', 'label', 'canDelete', 'show']),
+
+        tenantId: tenant.id,
+        createdById: currentUser.id,
+        updatedById: currentUser.id,
+      },
+      {
+        transaction,
+      },
+    )
+
+    return this.findById(record.id, options)
+  }
+
+  static async update(id, data, options: IRepositoryOptions) {
+    const currentUser = SequelizeRepository.getCurrentUser(options)
+
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const currentTenant = SequelizeRepository.getCurrentTenant(options)
+
+    let record = await options.database.memberAttributeSettings.findOne({
+      where: {
+        id,
+        tenantId: currentTenant.id,
+      },
+      transaction,
+    })
+
+    if (!record) {
+      throw new Error404()
+    }
+
+    record = await record.update(
+      {
+        ...lodash.pick(data, ['type', 'name', 'label', 'canDelete', 'show']),
+
+        updatedById: currentUser.id,
+      },
+      {
+        transaction,
+      },
+    )
+
+    return this.findById(record.id, options)
+  }
+
+  static async destroy(id, options: IRepositoryOptions) {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    const currentTenant = SequelizeRepository.getCurrentTenant(options)
+
+    const record = await options.database.memberAttributeSettings.findOne({
+      where: {
+        id,
+        tenantId: currentTenant.id,
+      },
+      transaction,
+    })
+
+    if (!record) {
+      throw new Error404()
+    }
+
+    if (record.canDelete) {
+      await record.destroy({
+        transaction,
+      })
+    }
+  }
+
+  static async findAndCountAll(
+    { filter, limit = 0, offset = 0, orderBy = '' },
```

Comment:
Same here - introduce a typescript interface so that whoever calls this function knows what is expected from `filter` parameter.